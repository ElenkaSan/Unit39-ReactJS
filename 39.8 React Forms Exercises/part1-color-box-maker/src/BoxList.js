import React, { useState } from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";
import { v4 as uuid } from "uuid";

const BoxList = () => {
    // const INITIAL_STATE = [
    //     { id: uuid(), width: 100, height: 85, color: "yellow" },
    //     { id: uuid(), width: 90, height: 80, color: "purple" }
    //   ]
    // const [lists, setLists] = useState(INITIAL_STATE);
    const [lists, setLists] = useState([]);

    const addBox = lists => {
        let newBox = { ...lists, id: uuid() };
        setLists(lists => [...lists, newBox]);
    };

    const removing = (idBox) => {
        setLists(lists.filter(({ id }) => id !== idBox));
    };
    
   return (
     <div>
          <h3>Create The Color Box</h3>
       <NewBoxForm addingBox={addBox} />
       {lists.map(({ id, width, height, color }) =>
       <Box remove={removing} key={id} id={id} width={width} height={height} color={color} />
       )}
     </div>
   )
};

export default BoxList;
