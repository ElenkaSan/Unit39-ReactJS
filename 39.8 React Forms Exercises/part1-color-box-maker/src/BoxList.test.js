import React from "react";
import { render, fireEvent, wait } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", function() {
  render(<BoxList />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add new box and delete", function () {
  const { queryByText, getByTestId } = render(<BoxList />);
  const heightInput = getByTestId('height');
  expect(heightInput).toBeEmpty; 
  const btn = queryByText('Add Box');
  fireEvent.change(heightInput, { target: { value: "350px" } });
  fireEvent.click(btn);
  expect(queryByText(`height: '350px'`)).toBeInTheDocument;
  
  const removeBtn = queryByText('X');
  fireEvent.click(removeBtn);
  expect(removeBtn).not.toBeInTheDocument();
  expect(heightInput).toBeEmpty; 
})
