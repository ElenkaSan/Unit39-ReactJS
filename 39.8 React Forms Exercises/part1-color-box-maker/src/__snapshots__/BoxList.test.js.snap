// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Create The Color Box
    </h3>
    <form>
      <label
        for="width"
      >
        Width: 
      </label>
      <input
        id="width"
        max="500"
        min="50"
        name="width"
        type="range"
        value=""
      />
      <label
        for="height"
      >
        Height: 
      </label>
      <input
        data-testid="height"
        id="height"
        max="500"
        min="50"
        name="height"
        type="range"
        value=""
      />
      <label
        for="color"
      >
        Color
      </label>
      <input
        id="color"
        name="color"
        type="text"
        value=""
      />
      <button
        id="newBox"
      >
        Add Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
